File: langs/python/pjs/isses/M001.py
# from functools import reduce
from itertools import combinations, permutations
# import numpy as np
# import sys
from collections import deque
import math

from tester import zz_act
from from_file import ffUser
from to_string import sSs, sNs, unwords
from from_string import to_i, to_is, to_nums

def m001(ss):
  first = ss[0].split(" "); second = ss[1].split(" ")
  return first + second
# def test(): ss = m001(["fishing gardening swimming fishing","hunting fishing fishing biting"]); print(ss)

def domain(ss):
  res = {}
  for s in ss:
    res[s] = res[s] + 1 if res.get(s) else 1
  return res
# def test(): res = domain(["fishing","gardening","swimming","fishing","hunting","fishing","fishing","biting"]); print(res)

def domains(ss): #main
  all = m001(ss); dom = domain(all); res = max(dom.values()); return [str(res)]
def test(): ans = domains(["fishing gardening swimming fishing","hunting fishing fishing biting"]); print(ans)

def user():
  res = []
  res.append(input().strip())
  res.append(input().strip())
  # lim = len(res[0])-1
  # for _ in range(lim): res.append(input().strip())
  return res

def product():
  u = user()
  # print(u)
  ans = domains(u)
  print("\n".join(ans))

def developer(issue, result, name):
  ans = domains(ffUser(issue).all()); exp = ffUser(result).all(); zz_act("M001 Test", sSs(ans),str(exp), name)
def develop():
  issue1 = "a:/pj/mamo/refs/isses/issue1.txt"; result1 = "a:/pj/mamo/refs/isses/result1.txt"
  issue2 = "a:/pj/mamo/refs/isses/issue2.txt"; result2 = "a:/pj/mamo/refs/isses/result2.txt"
  issue3 = "a:/pj/mamo/refs/isses/issue3.txt"; result3 = "a:/pj/mamo/refs/isses/result3.txt"
  issue4 = "a:/pj/mamo/refs/isses/issue4.txt"; result4 = "a:/pj/mamo/refs/isses/result4.txt"
  issue5 = "a:/pj/mamo/refs/isses/issue5.txt"; result5 = "a:/pj/mamo/refs/isses/result5.txt"
  issue6 = "a:/pj/mamo/refs/isses/issue6.txt"; result6 = "a:/pj/mamo/refs/isses/result6.txt"
  developer(issue1, result1, "test-1"); developer(issue2, result2, "test-2")
  developer(issue3, result3, "test-3"); developer(issue4, result4, "test-4")
  developer(issue5, result5, "test-5"); developer(issue6, result6, "test-6")

def refactor():
  issue = "a:/pj/mamo/refs/isses/issue7.txt"; result = "a:/pj/mamo/refs/isses/result7.txt"
  user = ffUser(issue).all(); exp = ffUser(result).all()
  # print(user); print(exp)
  ans = domains(user)
  zz_act("M001 Test", sSs(ans),str(exp), "refactor")

# main M001 m001 domain user
if __name__ == '__main__':
  # test()
  # refactor()
  develop()
  # product()

File: langs/ruby/pjs/isses/M001.rb
#encoding: UTF-8
require '../../utils/tester.rb'; require '../../utils/from_file.rb'
require '../../utils/from_string.rb'; require '../../utils/to_string.rb'
# fs -> to_nums; ts -> sNs;

def m001(ss)
  first = ss[0].split " "; second = ss[1].split " "; first + second;
end
# def test() res = m001 ["fishing gardening swimming fishing","hunting fishing fishing biting"]; p(res) end   #exp is exp
def domain(ss) res = Hash.new(0); for s in ss do res[s] += 1 end; res end
# def test() res = domain ["fishing","gardening","swimming","fishing","hunting","fishing","fishing","biting"]; p res end
def domains(ss) fs = m001 ss; mp = domain fs; res = mp.values.max; [res.to_s] end #main
def test()  res = domains ["fishing gardening swimming fishing","hunting fishing fishing biting"]; p res end

def user()
  res = [gets.strip!]
  res << gets.strip!;
  # lim = res[0].size() - 1
  # lim.times{ res << gets.strip!}; 
  return res
end

def product()
  u = user;
  # p u
  res = domains(u); puts (res.join("\n"))
end

def developer(issue,result,name)
  ans = domains((ffUser issue).all()); exp = (ffUser result).all(); zz_act "M001 Test", sNs(ans), exp, name
end
def develop() 
  issue1 = "a:/pj/mamo/refs/isses/issue1.txt"; result1 = "a:/pj/mamo/refs/isses/result1.txt"
  issue2 = "a:/pj/mamo/refs/isses/issue2.txt"; result2 = "a:/pj/mamo/refs/isses/result2.txt"
  issue3 = "a:/pj/mamo/refs/isses/issue3.txt"; result3 = "a:/pj/mamo/refs/isses/result3.txt"
  issue4 = "a:/pj/mamo/refs/isses/issue4.txt"; result4 = "a:/pj/mamo/refs/isses/result4.txt"
  issue5 = "a:/pj/mamo/refs/isses/issue5.txt"; result5 = "a:/pj/mamo/refs/isses/result5.txt"
  issue6 = "a:/pj/mamo/refs/isses/issue6.txt"; result6 = "a:/pj/mamo/refs/isses/result6.txt"
  developer(issue1,result1,"test-1"); developer(issue2,result2,"test-2");
  developer(issue3,result3,"test-3"); developer(issue4,result4,"test-4");
  developer(issue5,result5,"test-5"); developer(issue6,result6,"test-6");
end
def refactor()
  issue = "a:/pj/mamo/refs/isses/issue1.txt"; result = "a:/pj/mamo/refs/isses/result1.txt"
  # ans = (ffUser issue).all; exp = (ffUser result).all; p ans; p exp;
  ans = domains ((ffUser issue).all); exp = (ffUser result).all;
  zz_act "M001 Test", sNs(ans), exp, "refactor"
end

# main M001 m001 domain user
if __FILE__ == $0 then
  # test
  # refactor
  develop
  # product
end 